App/Dependencies/ExposureDetection/ImmuniTemporaryExposureKeyProvider.swift

```diff
@@ -49,18 +57,36 @@ class ImmuniTemporaryExposureKeyProvider: TemporaryExposureKeyProvider {
       .delete(chunks.flatMap { $0.localUrls })
   }
 
-  private func getMissingChunksIndexes(latestKnownChunkIndex: Int?) -> Promise<[Int]> {
+  func getMissingChunksIndexes(latestKnownChunkIndex: Int?) -> Promise<[Int]> {
     self.networkManager.getKeysIndex()
       .then { (keysIndex: KeysIndex) -> [Int] in
-        guard let latestKnownChunkIndex = latestKnownChunkIndex else {
-          return Array(keysIndex.oldest ... keysIndex.newest)
-        }
 
-        guard keysIndex.newest > latestKnownChunkIndex else {
+        let latestKnown = latestKnownChunkIndex ?? -1
+
+        guard keysIndex.newest > latestKnown else {
```

Comment:
Do we also need to check whether `newest` is always greater than `oldest` or we just always trust the server?